### ISSUE TITLE
[$250] Android - 3-button navigation bar overlaps the Copilot account

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#53842]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v9.0.73-6
**Reproducible in staging?:** Yes
**Reproducible in production?:** Yes
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** Yes, reproducible on both
**Email or phone of affected tester (no customers):** testpayment935+780@gmail.com
**Issue reported by:** Applause Internal Team

## Action Performed:
Prerequire - Make sure that your Android device uses 3-button navigation bar
Steps:
1. Navigate to hybrid app in android mobile (Account A) and https://staging.new.expensify.com in other device(Account B)
2. Account B add a copilot to the account (account A)
3. In app - Log in to Account A.
4. Go to settings and click the account switcher

## Expected Result:
Android 3-button navigation bar doesn't overlap the copilot account

## Actual Result:
Android 3-button navigation bar overlaps the copilot account

## Workaround:
Unknown

## Platforms:
- [x] Android: Standalone
- [x] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/ac4a9ca3-cbe9-4d1c-9173-e91830a92ad7

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021869485429092398114</li>
        <li>Upwork Job ID: 1869485429092398114</li>
        <li>Last Price Increase: 2024-12-18</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @jayeshmangwani</details>

**Comment by ChavdaSachin** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Android - 3-button navigation bar overlaps the Copilot account

### What is the root cause of that problem?
A case was missed during android edge-to-edge implementation

### What changes do you think we should make in order to solve the problem?
Change 
```js
                    shouldUseModalPaddingStyle={false}
                    innerContainerStyle={{paddingBottom: unmodifiedPaddings.bottom}}
```
https://github.com/Expensify/App/blob/9975b27ac58d2dd78a521064fd993ed5de097772/src/components/AccountSwitcher.tsx#L203

Like it is implemented here

https://github.com/Expensify/App/blob/9975b27ac58d2dd78a521064fd993ed5de097772/src/pages/Search/SearchTypeMenuNarrow.tsx#L215-L216


### What specific scenarios should we cover in automated tests to prevent reintroducing this issue in the future?

NA
### What alternative solutions did you explore? (Optional)

**Reminder:** Please use plain English, be brief and avoid jargon. Feel free to use images, charts or pseudo-code if necessary. Do not post large multi-line diffs or write walls of text. Do not create PRs unless you have been hired for this job.

**Comment by OfstadC** [comment]
Will test this this afternoon

**Comment by OfstadC** [comment]
I didn't get a chance to test this yet - will tackle on Tuesday when i'm back in office

**Comment by OfstadC** [comment]
Not able to reproduce
<img width="406" alt="image" src="https://github.com/user-attachments/assets/913a0139-b45e-456c-ad16-a49e5e9fed65" />

I tried on 4 different devices with the 3 button navigation and all copilot accounts were visible to select

**Comment by ChavdaSachin** [comment]
I was able to repro this with 6 copilots and pixel 6 emulator

**Comment by izarutskaya** [comment]
Reproducible in latest build v9.0.76-12

![Screenshot_20241218_184948_Expensify](https://github.com/user-attachments/assets/60c88b49-6547-4853-a285-81e56e66024d)

**Comment by jayeshmangwani** [comment]
@ChavdaSachin 's [Proposal](https://github.com/Expensify/App/issues/53842#issuecomment-2531375603) looks good and worked well while testing; let's go with it.

🎀 👀 🎀 C+ reviewed

**Comment by jayeshmangwani** [comment]
@daledah I couldn’t see much of a difference locally. I applied the suggested changes and reviewed it again, but I’m unsure what major differences to look for. Could you add a comparison video for clarity?

**Comment by jayeshmangwani** [comment]
> You can see the different in my images above.

 I can’t see the difference locally; could you add a full video instead of just a cropped screenshot?

**Comment by jayeshmangwani** [comment]
@daledah Are you sure you’ve applied these changes?
I’ve tried a few times, and it’s working fine for me.
![Uploading Screenshot 2024-12-20 at 1.17.04 PM.png…]()

**Comment by jayeshmangwani** [comment]
@ChavdaSachin, could you please confirm if your proposal might break the ScrollView padding based on this [video](https://github.com/Expensify/App/issues/53842#issuecomment-2556451511)?

**Comment by jayeshmangwani** [comment]
https://github.com/user-attachments/assets/2367aa52-0c5c-4316-accd-81a463eedffb

**Comment by ChavdaSachin** [comment]
[Updated Proposal](https://github.com/Expensify/App/issues/53842#issuecomment-2531375603)
@jayeshmangwani yes, my old proposal breaks the style on web, But I have updated proposal.
thanks @daledah.

**Comment by jayeshmangwani** [comment]
>  I saw in your video scrollContainerStyle is still existed.

I’ve already added the code changes in my [previous comment](https://github.com/Expensify/App/issues/53842#issuecomment-2556460393), @daledah, for confirmation. And you mentioned that you have the same code changes! 😄 and you actually [confirmed](https://github.com/Expensify/App/issues/53842#issuecomment-2556465286) that you applied same changes

**Comment by jayeshmangwani** [comment]
@youssef-lr Based on this [comment](https://github.com/Expensify/App/issues/53842#issuecomment-2556392204), it seems the selected proposal had an issue that I overlooked. We can go ahead with @daledah 's [proposal](https://github.com/Expensify/App/issues/53842#issuecomment-2533615718) instead.

**Comment by jayeshmangwani** [comment]
No issues—my bad! Sorry, I just misinterpreted the previously selected proposal.

**Comment by ChavdaSachin** [comment]
And also for the device with navigation buttons, padding is overwhelming.
Comparison BW my updated proposal and the other proposal.
![Screenshot 2024-12-20 at 3 05 52 PM](https://github.com/user-attachments/assets/98bb2daf-7dc8-467e-ba5e-7235e88d5d8e)
![Screenshot 2024-12-20 at 3 06 05 PM](https://github.com/user-attachments/assets/f6e64037-c6c2-4173-aa44-f924400c44f0)

**Comment by jayeshmangwani** [comment]
Thanks for the [comparison screenshots](https://github.com/Expensify/App/issues/53842#issuecomment-2556623042) @ChavdaSachin , 
@youssef-lr Do we need to update (reduce) the padding of the popover in this issue to make it consistent with the other bottom-deck popovers?

**Comment by Pujan92** [comment]
Seeing a similar issue https://github.com/Expensify/App/issues/53779

**Comment by jayeshmangwani** [comment]
I [agree](https://github.com/Expensify/App/issues/53842#issuecomment-2563023041) with @Pujan92 . If we can address the Android 3-button navigation bar overlaps in a single PR, it would be more efficient than creating separate issues and PRs for similar cases. Handling these related issues together in one PR would save time and effort

@youssef-lr, what do you think about merging these similar issues https://github.com/Expensify/App/issues/53779 and https://github.com/Expensify/App/issues/53650 into a single issue?

**Comment by jayeshmangwani** [comment]
@youssef-lr Whenever you get a chance, please look at this comment https://github.com/Expensify/App/issues/53842#issuecomment-2563579966 .

**Comment by jayeshmangwani** [comment]
@mountiny @kirillzyusko Tagging you here for visibility of this issue too, so we can address similar issues [holistically](https://github.com/Expensify/App/issues/53650#issuecomment-2576206834) if needed

**Comment by mountiny** [comment]
Yes please lets wait and handle it here https://expensify.slack.com/archives/C049HHMV9SM/p1736360325497079?thread_ts=1736238279.772929&cid=C049HHMV9SM

**Comment by OfstadC** [comment]
Looks like this is being [discussed](https://expensify.slack.com/archives/C049HHMV9SM/p1736360325497079?thread_ts=1736238279.772929&cid=C049HHMV9SM) in Slack

**Comment by OfstadC** [comment]
No update

**Comment by jayeshmangwani** [comment]
Not Overdue, Issue being discussed here https://expensify.slack.com/archives/C049HHMV9SM/p1736360325497079?thread_ts=1736238279.772929&cid=C049HHMV9SM

**Comment by jayeshmangwani** [comment]
Not overdue, and no new update on the Slack thread.

**Comment by jayeshmangwani** [comment]
There are a few updates on the Slack thread, and it looks like Chris is handling all similar issues. Tagging you, @chrispader, so you can track this issue while addressing the translucent issue holistically.

**Comment by OfstadC** [comment]
Any update @jayeshmangwani ? 😃

**Comment by jayeshmangwani** [comment]
@chrispader, is there an ongoing PR for these [Slack](https://expensify.slack.com/archives/C049HHMV9SM/p1736360325497079?thread_ts=1736238279.772929&cid=C049HHMV9SM) changes that we can track?

**Comment by jayeshmangwani** [comment]
@mountiny, I’m tagging you here based on the Slack thread about fixing the bottom padding for Android. 

Do you have any idea which PR is tackling the 3-button navigation issue? I see [this PR](https://github.com/Expensify/App/pull/55633), but it’s been in draft for a month

**Comment by jayeshmangwani** [comment]
@OfstadC, can we please make this Issue weekly until we know, for which PR we’re waiting on?

**Comment by mountiny** [comment]
@jayeshmangwani That is correct PR I think, Margelo had limited hours, but I hope they can speed things up now

**Comment by OfstadC** [comment]
Thanks @mountiny -  i'll leave this is Daily for now unless you think it should be moved to weekly 😃

**Comment by jayeshmangwani** [comment]
Monitoring this PR https://github.com/Expensify/App/pull/55633 , same as last update https://github.com/Expensify/App/issues/53842#issuecomment-2663249988

**Comment by jayeshmangwani** [comment]
No new updates on the PR from [this update](https://github.com/Expensify/App/issues/53842#issuecomment-2676776722).

**Comment by jayeshmangwani** [comment]
Unfortunately, no new updates here 😄.

**Comment by chrispader** [comment]
@jayeshmangwani [this PR](https://github.com/Expensify/App/pull/57181) is going to fix this issue.

The [Base PR](https://github.com/Expensify/App/pull/57128) should hopefully be merged soon. https://github.com/Expensify/App/pull/57181 is going to migrate all modals to the new bottom safe area handling, which will also fix your issue! 👍🏼

Actively working on this now, since the other PRs is already reviewed...

**Comment by chrispader** [comment]
@jayeshmangwani this is fixed in https://github.com/Expensify/App/pull/57128

|  Soft keys |  Gesture bar |
|--------|---------|
| <img src="https://github.com/user-attachments/assets/e3a2fa0c-30b0-4cff-aa3d-354d061d755a" /> | <img src="https://github.com/user-attachments/assets/d72ea9fb-0d4e-4576-a556-efcebea19806" /> |

**Comment by chrispader** [comment]
> [@chrispader](https://github.com/chrispader) Thanks for keeping us updated! I'll keep an eye on this PR [#57128](https://github.com/Expensify/App/pull/57128). Once it hits staging, we can ask QA to retest the issue.

@jayeshmangwani, sorry my fault. https://github.com/Expensify/App/pull/5718 is going to fix this issue. The Base PR is not yet going to change anything UI-related.

**Comment by jayeshmangwani** [comment]
Monitoring https://github.com/Expensify/App/pull/57181

**Comment by jayeshmangwani** [comment]
Same as [above](https://github.com/Expensify/App/issues/53842#issuecomment-2714800319)

**Comment by jayeshmangwani** [comment]
No updates on the PR yet.

**Comment by jayeshmangwani** [comment]
Same as above.

**Comment by jayeshmangwani** [comment]
Watching the linked PR—no updates so far.

**Comment by jayeshmangwani** [comment]
Same as [above](https://github.com/Expensify/App/issues/53842#issuecomment-2774638298)

**Comment by jayeshmangwani** [comment]
Same https://github.com/Expensify/App/issues/53842#issuecomment-2802926095

**Comment by jayeshmangwani** [comment]
PR is still in draft.

**Comment by chrispader** [comment]
I hope to finish it up by the end of the week!

**Comment by jayeshmangwani** [comment]
Same as above https://github.com/Expensify/App/issues/53842#issuecomment-2841492606

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by jayeshmangwani** [comment]
issue isn’t reproducible now

![Image](https://github.com/user-attachments/assets/d2f54391-932c-435d-bedf-5544d86f4816)

**Comment by jayeshmangwani** [comment]
@OfstadC Since the issue no longer reproduces https://github.com/Expensify/App/issues/53842#issuecomment-2868308064 https://github.com/Expensify/App/issues/53842#issuecomment-2869530507, are we good to close this issue?

**Comment by OfstadC** [comment]
Yes! Let's close it out! 😃
